smack-websocket/src/main/java/org/jivesoftware/smack/websocket/implementations/okhttp/OkHttpWebsocketImpl.java

```diff
@@ -0,0 +1,190 @@
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jivesoftware.smack.c2s.internal.ModularXmppClientToServerConnectionInternal;
+import org.jivesoftware.smack.packet.StreamClose;
+import org.jivesoftware.smack.packet.TopLevelStreamElement;
+import org.jivesoftware.smack.util.PacketParserUtils;
+import org.jivesoftware.smack.websocket.LoggingInterceptor;
+import org.jivesoftware.smack.websocket.XmppWebsocketTransportModule.XmppWebsocketTransport.DiscoveredWebsocketEndpoints;
+import org.jivesoftware.smack.websocket.elements.CloseElement;
+import org.jivesoftware.smack.websocket.elements.OpenElement;
+import org.jivesoftware.smack.websocket.implementations.AbstractWebsocket;
+import org.jivesoftware.smack.websocket.rce.WebsocketRemoteConnectionEndpoint;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+
+public final class OkHttpWebsocketImpl extends AbstractWebsocket {
+
+    private static final Logger LOGGER = Logger.getLogger(OkHttpWebsocketImpl.class.getName());
+    private static OkHttpClient okHttpClient = null;
+
+    private boolean closeInitiatedByClient = false;
+    private boolean closeInititatedByServer = false;
+    private String openStreamHeader;
+    private final ModularXmppClientToServerConnectionInternal connectionInternal;
+    private WebSocket currentWebsocket;
+    private WebsocketConnectionPhase phase;
+    private Map<WebsocketRemoteConnectionEndpoint, Throwable> failureMap = new HashMap<>();
+
+    public OkHttpWebsocketImpl(ModularXmppClientToServerConnectionInternal connectionInternal,
+            DiscoveredWebsocketEndpoints discoveredWebsocketEndpoints) {
+        this.connectionInternal = connectionInternal;
+
+        if (okHttpClient == null) {
+            // Creates an instance of okHttp client.
+            buildWebsocketClient();
+        }
+    }
+
+    private static void buildWebsocketClient() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        // Add some mechanism to enable and disable this intercepter.
+        builder.addInterceptor(new LoggingInterceptor());
+        okHttpClient = builder.build();
+    }
+
+    @Override
+    public void connect(WebsocketRemoteConnectionEndpoint endpoint) throws Throwable {
+        final String currentUri = endpoint.getWebsocketEndpoint();
+        Request request = new Request.Builder()
+                              .url(currentUri)
+                              .header("Sec-WebSocket-Protocol", "xmpp")
+                              .build();
+
+        WebSocketListener listener = new WebSocketListener() {
+            @Override
+            public void onOpen(WebSocket webSocket, Response response) {
+                LOGGER.log(Level.INFO, "Websocket is open");
```

Comment:
This should not be of level `INFO`. Make it `FINER` and add the response to the log string.